### PR TITLE
Include certificate expiry check in all kubernetes commands

### PR DESCRIPTION
**Pull request by pravinpushkar**
Signed-off-by: Pravin Pushkar <ppushkar@microsoft.com>

# Description

Show warning for certificate expiry on all kubernetes related commands

## Issue reference

Please reference the issue this PR will close: #893 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: pkg/kubernetes/mtls.go
Comment (mukundansundar) [inline pkg/kubernetes/mtls.go:113]:
```suggestion
func CheckForCertExpiry() {
```

**Review by mukundansundar — COMMENTED** [review]
@pravinpushkar Can you take a look at the k8s E2E test. All of them seem to be failing.

File: pkg/kubernetes/mtls.go
Comment (shubham1172) [inline pkg/kubernetes/mtls.go:120]: nit: Should we keep 30 as a constant above?

Comment (mukundansundar) [reply pkg/kubernetes/mtls.go:120]: no magic numbers please ... Put it as constants above in code ...

**Comment by pravinpushkar** [comment]
> @pravinpushkar Can you take a look at the k8s E2E test. All of them seem to be failing.

Running now.

File: pkg/kubernetes/mtls.go
Comment (shubham1172) [inline pkg/kubernetes/mtls.go:118]: nit: Usually it's easier to read go code that goes like this:
```go
if err != nil {
    return
}
// other logic
```

Comment (shubham1172) [reply pkg/kubernetes/mtls.go:118]: Reference https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

File: pkg/kubernetes/mtls.go
Comment (shubham1172) [inline pkg/kubernetes/mtls.go:122]: Same nit for line 120 as above.
For if/else checks from line 122, we can try using a switch case here for readability.

File: pkg/kubernetes/mtls.go
Comment (shubham1172) [inline pkg/kubernetes/mtls.go:132]: There are colors defined in the `print` pkg which look unused. As a caller of `WarningStatusEvent`, I would rely on the function to set the colors instead. It can use different colors based on the OS, environment, accessibility guidelines, etc., but the caller should remain agnostic. What are your thoughts, @pravinpushkar, @mukundansundar?

File: pkg/kubernetes/mtls.go
Comment (shubham1172) [inline pkg/kubernetes/mtls.go:131]: nit: `WarningStatusEvent` takes variadic params and uses `fmt.Sprintf` internally. We can also do something like
```go
print.WarningStatusEvent(os.Stdout, "%s. Expiry date: %s.\n Kindly renew to avoid any service interruptions.", warningMessage, expiry.Format(time.RFC1123))
```

File: pkg/kubernetes/mtls.go
Comment (mukundansundar) [reply pkg/kubernetes/mtls.go:132]: the caller should remain agnostic of the colour ... Ideally only the behaviour that is wanted ... The colour will be set internally as defined by the function and needs to be consistent across all such calls.

Comment (shubham1172) [reply pkg/kubernetes/mtls.go:132]: Thanks for sharing your thoughts. For this PR maybe we can omit the color part then, and track the behavior of `print` pkg separately? I see color being used in only one more place that might need refactoring.

https://github.com/dapr/cli/blob/f89be62068c39ffd3492ab24025e3c0624a56c61/pkg/standalone/standalone.go#L788

Comment (pravinpushkar) [reply pkg/kubernetes/mtls.go:132]: I am also ok to remove the color. Just a thought,  normally people use blackish/whitish background terminal and in both these theme yellow(it is not actually yellow, it is kind of orange) would fit nicely at the same time we do want to highlight this because of the seriousness.

File: pkg/kubernetes/mtls.go
Comment (pravinpushkar) [reply pkg/kubernetes/mtls.go:122]: I have mixed feelings here. Normally switch cases are good if we compare with constants like.  I might be wrong though. Thoughts ?
```
n=5
switch n {
case 1:
case 2:
default:
}
```

File: pkg/kubernetes/mtls.go
Comment (pravinpushkar) [reply pkg/kubernetes/mtls.go:132]: It will be better to have diff PR. I will remove the color part.

File: pkg/kubernetes/mtls.go
Comment (shubham1172) [reply pkg/kubernetes/mtls.go:131]: The only benefit IMO is that this looks more idiomatic. I can see longer lines in our code-base, so might just want to keep it that way.  Or, we can introduce line breaks in the same statement (and where to put a line break can be subjective, for example, https://github.com/mvdan/gofumpt/issues/2). I would suggest this (the two function params are `io.Writer` and `...args`):
```go
print.WarningStatusEvent(os.Stdout, 
		"%s. Expiry date: %s.\n Kindly renew to avoid any service interruptions.", warningMessage, expiry.Format(time.RFC1123))
```

File: pkg/kubernetes/mtls.go
Comment (shubham1172) [reply pkg/kubernetes/mtls.go:122]: Sure @pravinpushkar, it's only a nit. Go does support conditional switch case though:
```go
switch {
case x < 0:
        // do something
case x > 0:
        // do something
case x == 0:
        // do something
}
```

**Review by mukundansundar — COMMENTED** [review]
LGTM. The only concern that I have is, is it possible to have a temporary cert created following steps [here](https://docs.dapr.io/operations/security/mtls/#bringing-your-own-certificates-1) and have that verified as part of E2E ?
@berndverst Can you post your comments on this?

File: pkg/kubernetes/mtls.go
Comment (berndverst) [inline pkg/kubernetes/mtls.go:126]:
```suggestion
			warningMessage = "Dapr root certificate of your Kubernetes cluster expires today."
```

File: pkg/kubernetes/mtls.go
Comment (berndverst) [inline pkg/kubernetes/mtls.go:128]:
```suggestion
			warningMessage = "Dapr root certificate of your Kubernetes cluster has expired."
```

File: pkg/kubernetes/mtls.go
Comment (berndverst) [inline pkg/kubernetes/mtls.go:130]:
```suggestion
			warningMessage = fmt.Sprintf("Dapr root certificate of your Kubernetes cluster expires in %v days", daysRemaining)
```

File: pkg/kubernetes/mtls.go
Comment (berndverst) [inline pkg/kubernetes/mtls.go:132]:
```suggestion
		helpMessage := "Please see docs.dapr.io for certificate renewal instructions to avoid service interruptions."
```

**Review by berndverst — APPROVED** [review]
Looks good to me assuming the various strings are updated.

I'm not an approve or maintainer of the CLI so @artursouza or @mukundansundar will need to review and merge after you make the update. You can just accept the changes I suggested (but that may require you to manually fix the DCO check).

**Comment by berndverst** [comment]
> LGTM. The only concern that I have is, is it possible to have a temporary cert created following steps [here](https://docs.dapr.io/operations/security/mtls/#bringing-your-own-certificates-1) and have that verified as part of E2E ?
> @berndverst Can you post your comments on this?

@mukundansundar I don't have concerns here. The existing `dapr mtls expiry` command works with certificates generated by Dapr and certificates manually generated and provided per our docs. This PR here uses the same logic for getting the expiry.

We could create an E2E test for manually generating certificates, deploying them to Kubernetes and verifying with this command -- in fact that's what I did during the last community call (but manually so). I don't feel strongly that we need this. The `mtls expiry` command is well proven at this point.
I'll leave it up to you as the CLI maintainer.

This PR looks good to me assuming we change the various returned strings - they are a bit clumsy right now.

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**
